esp/public/media/scripts/program/modules/scheduling_checks.jsx

```diff
@@ -114,3 +148,71 @@ var RefreshButton = React.createClass({
     </button>;
   },
 });
+
+
+// Modified from react-json-table example code.
+// Required props: rows, saveState (with sort and greyed attributes), settings (with header attribute)
+var SelectTable = React.createClass({
+  getInitialState: function(){
+    // We will store the sorted column and whether each row is greyed out
+    return {sort: this.props.saveState.sort, greyed : this.props.saveState.greyed};
+  },  
+  render: function(){
+    var me = this,
+        // clone the rows
+        items = this.props.rows.slice()
+    ;
+    // Sort the table
+    if( this.state.sort ){
+      items.sort( function( a, b ){
+         return a[ me.state.sort ] > b[ me.state.sort ] ? 1 : -1;
+      });
+    }
+      
+    if (this.props.columns == undefined) {
+      return <JsonTable 
+        rows={items} 
+        settings={ this.getSettings() } 
+        onClickHeader={ this.onClickHeader }
+        onClickRow={ this.onClickRow }
+      />;
+    } else {
+      return <JsonTable 
+        rows={items} 
+        columns={this.props.columns}
+        settings={ this.getSettings() } 
+        onClickHeader={ this.onClickHeader }
+        onClickRow={ this.onClickRow }
+      />;
+    }
+  },
+  
+  getSettings: function(){
+      var me = this;
+      // We will add some classes to the selected rows and cells
+      return {
+        headerClass: function( current, key ){
+            if( me.state.sort == key )
```

Comment:
I'm a fan of explicit else clauses, but we don't have a style guide so you can ignore me if you want.